src/main/java/edu/ie3/datamodel/io/connectors/SqlConnector.java

```diff
@@ -115,21 +121,56 @@ public void shutdown() {
   }
 
   /**
-   * Extracts all field to value maps from the ResultSet, one for each row
+   * Method to execute a {@link PreparedStatement} and return its result as a stream.
    *
-   * @param rs the ResultSet to use
-   * @return a list of field maps
+   * @param ps to execute
+   * @param fetchSize used for {@link PreparedStatement#setFetchSize(int)}
+   * @return a stream of maps
+   * @throws SQLException if an exception occurred while executing the query
    */
-  public List<Map<String, String>> extractFieldMaps(ResultSet rs) {
-    List<Map<String, String>> fieldMaps = new ArrayList<>();
-    try {
-      while (rs.next()) {
-        fieldMaps.add(extractFieldMap(rs));
+  public Stream<Map<String, String>> toStream(PreparedStatement ps, int fetchSize)
+      throws SQLException {
+    ps.setFetchSize(fetchSize);
+    ResultSet resultSet = ps.executeQuery();
+    Iterator<Map<String, String>> sqlIterator = getSqlIterator(resultSet);
+
+    return StreamSupport.stream(
+            Spliterators.spliteratorUnknownSize(
+                sqlIterator, Spliterator.NONNULL | Spliterator.IMMUTABLE),
+            true)
+        .onClose(
+            () -> {
+              try (resultSet) {
+                log.debug("Resources successfully closed.");
+              } catch (SQLException e) {
+                log.warn("Failed to properly close sources.", e);
+              }
+            });
+  }
+
+  /**
+   * Returns an {@link Iterator} for the given {@link ResultSet}.
+   *
+   * @param rs given result set
+   * @return an iterator
+   */
+  public Iterator<Map<String, String>> getSqlIterator(ResultSet rs) {
+    return new Iterator<>() {
+      @Override
+      public boolean hasNext() {
+        try {
+          return rs.next();
+        } catch (SQLException e) {
+          log.error("Exception at extracting next ResultSet: ", e);
+          return false;
+        }
       }
-    } catch (SQLException e) {
-      log.error("Exception at extracting ResultSet: ", e);
-    }
-    return fieldMaps;
+
+      @Override
+      public Map<String, String> next() {
+        return extractFieldMap(rs);
```

Comment:
Also here. Maybe `extractFieldMap` should not catch the SQLException, but we handle it inside the `next()` method.